scripts/build-rhcc-image.sh

```diff
@@ -83,24 +97,54 @@ else
     INTERNAL_IMAGE_NAME=cb-rhcc/sync-gateway
 fi
 
+# Figure out whether which platforms we're targeting
+case ${PRODUCT} in
+    couchbase-server)
+        if version_lte ${VERSION} 7.2.0; then
+            arches="amd64"
+            platforms="linux/amd64"
+        else
+            arches="amd64 arm64"
+            platforms="linux/amd64,linux/arm64"
+        fi
+        ;;
+    sync-gateway)
+        if version_lte ${VERSION} 3.0.4; then
+            arches="amd64"
+            platforms="linux/amd64"
+        else
+            arches="amd64 arm64"
+            platforms="linux/amd64,linux/arm64"
+        fi
+        ;;
+esac
+
+
 # Build and push images
 for registry in ghcr.io build-docker.couchbase.com; do
     IMAGE=${registry}/${INTERNAL_IMAGE_NAME}:${VERSION}-${BLD_NUM}
 
     ${SCRIPT_DIR}/update-base.sh ${DOCKERFILE}
-    echo @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
-    echo Building ${IMAGE}
-    echo @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
-    docker build ${CACHE_ARG} \
-      --build-arg PROD_VERSION=${VERSION} \
-      --build-arg STAGING=${STAGING} \
-      -f ${DOCKERFILE} -t ${IMAGE} .
-
-    if [ "x${DRYRUN}" != "xyes" ]; then
+    if [ "${DRYRUN}" = "yes" ]; then
+        # For dry run, we build each architecture's image individually
+        # so we can load them into the local image store for testing
+        for arch in $arches; do
+            IMAGE=${registry}/${INTERNAL_IMAGE_NAME}:${VERSION}-${BLD_NUM}-${arch}
+            echo @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+            echo Building ${IMAGE}
+            echo @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+            docker buildx build --platform linux/${arch} --load ${CACHE_ARG} \
```

Comment:
So this is slightly problematic ... to support the older RHEL7-based releases, this job still runs on the "openshift" agent when building those Dockerfile.old versions (including Server 6.6.6). The openshift agent is is RHEL7-based. It certainly doesn't have and will never have "docker buildx build".

I guess we could tag the current tip of this redhat-openshift repository, and try to have the job check out that tag when building "historic" versions?

Reply:
I've brought `docker build` back into the mix for non-multiarch versions.